### ISSUE TITLE
touch bar events tricky thing

### ISSUE THREAD

**Issue by alexey-sh** [electron/electron#9421]
Hi
Just want to ask you to add a note that
`require('electron').remote.getCurrentWindow().removeAllListeners()`
will disable click events on touch bar buttons.

I mean this click event:
```
new TouchBarButton({
        label: 'Click',
        click: function () {
              console.log('clicked');
        }
});
```

thanks

**Comment by MarshallOfSound** [comment]
@alexey-sh That code will remove a lot more than just the touch bar events, it will remove a whole bunch of internal Electron events as well.  You shouldn't ever call `removeAllListeners` if you weren't personally responsible for adding all the listeners to that object, it will cause unintended side effects like this. Things like our internal `blur` listeners and such will also be removed.

I would say adding docs to say just don't use `removeAllListeners` on a browser window would be a good recommendation 👍

**Comment by sindresorhus** [comment]
> I would say adding docs to say just don't use removeAllListeners on a browser window would be a good recommendation 👍

I would take it even further. Either remove the method from the window or make it apply only to user event handles.

**Comment by az-chris** [comment]
Think I added an appropriate update to the documentation to address the issue.

**Comment by MarshallOfSound** [comment]
Fixed in #13229
